buildSrc/subprojects/integration-testing/src/main/kotlin/org/gradle/gradlebuild/test/integrationtests/DistributionTestingPlugin.kt

```diff
@@ -78,13 +79,14 @@ class DistributionTestingPlugin : Plugin<Project> {
     }
 
     private
-    fun DistributionTest.configureGradleTestEnvironment(providers: ProviderFactory, layout: ProjectLayout, basePluginConvention: BasePluginConvention) {
+    fun DistributionTest.configureGradleTestEnvironment(providers: ProviderFactory, layout: ProjectLayout, basePluginConvention: BasePluginConvention, objects: ObjectFactory) {
```

Comment:
Do we still need the `layout` here? Can we use the `ObjectFactory` to create `providers`, too?

Reply:
Right, seems we've added the directory/file property methods to `ObjectFactory` now.

Reply:
We need `layout` for `layout.projectDirectory` and `layout.buildDirectory` and `ObjectFactory` cannot create `Providers`.